src/librustc_middle/ty/context.rs

```diff
@@ -1181,6 +1179,49 @@ impl<'tcx> TyCtxt<'tcx> {
         }
     }
 
+    /// Constructs a `TyKind::Error` type and registers a `delay_span_bug` to ensure it gets used.
+    #[track_caller]
+    pub fn err(self) -> Ty<'tcx> {
+        self.err_with_message_and_location(
```

Comment:
Note that `#[track_caller]` will propagate through all functions in the call chain, so you can just call `err_with_message`

Reply:
Ah, I didn't know that! Thanks. And now I think I understand @eddyb's comment above.